src/Apps/Purchase/Components/PurchaseHistory.tsx

```diff
@@ -78,59 +80,149 @@ const OrderRow = (props: OrderRowProps) => {
   )
 }
 
+const PAGE_SIZE = 5
+
+const loadNext = (pageInfo, relay, setLoading) => {
+  const { hasNextPage, endCursor } = pageInfo
+
+  if (hasNextPage) {
+    this.loadAfter(endCursor, relay, setLoading)
+  }
+}
+
+const loadAfter = (cursor, relay, setLoading) => {
+  setLoading(true)
+
+  relay.refetch(
+    {
+      first: PAGE_SIZE,
+      after: cursor,
+      before: null,
+      last: null,
+    },
+    null,
+    error => {
+      setLoading(false)
+
+      if (error) {
+        console.error(error)
+      }
+    }
+  )
+}
 interface PurchaseHistoryProps {
-  orders: PurchaseHistory_orders
+  me: PurchaseHistory_me
+  relay: RelayRefetchProp
 }
 
 const PurchaseHistory: React.FC<PurchaseHistoryProps> = (
   props: PurchaseHistoryProps
 ) => {
-  const { orders } = props
-  const myOrders = orders.edges && orders.edges.map(x => x.node)
-  return (
+  const [loading, setLoading] = useState(false)
+  const { me } = props
+  const pageInfo = me.orders.pageInfo
+  const myOrders = me.orders.edges && me.orders.edges.map(x => x.node)
+  return !loading ? (
     <Box px={1}>
       <Serif size="5">Purchases</Serif>
       {myOrders.length ? (
         myOrders.map(order => <OrderRow key={order.code} order={order} />)
       ) : (
         <Sans size="2">No Orders</Sans>
       )}
+      <LargePagination
```

Comment:
🐈 An improvement (not in this PR!), could be to move `Pagination` stuff out of this component and up a level, so that the list component (this one), can re-render, have loading states, etc. - without disrupting the pagination bar.